src/os/filestore/DBObjectMap.cc

```diff
@@ -237,7 +237,7 @@ int DBObjectMap::DBObjectMapIteratorImpl::init()
       assert(0);
       return -EINVAL;
     }
-    parent_iter.reset(new DBObjectMapIteratorImpl(map, parent));
+    parent_iter = std::make_shared<DBObjectMapIteratorImpl>(map, parent);
```

Comment:
The advantage of make_shared() is that it combines the two allocations (the pointer itself, and the shared internal refcount/bookkeeping structure) into one. shared_ptr::reset() still needs to allocate this shared refcount structure, so the make_shared() optimization helps here too.